x/rollapp/keeper/msg_server.go

```diff
@@ -15,3 +20,40 @@ func NewMsgServerImpl(keeper Keeper) types.MsgServer {
 }
 
 var _ types.MsgServer = msgServer{}
+
+func (k msgServer) TriggerGenesisEvent(goCtx context.Context, msg *types.MsgRollappGenesisEvent) (*types.MsgRollappGenesisEventResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// Get the sender and validate they are in the whitelist
+	if !k.IsAddressInDeployerWhiteList(ctx, msg.Address) {
+		return nil, sdkerrors.ErrUnauthorized
+	}
+
+	// Get the rollapp
+	rollapp, found := k.GetRollapp(ctx, msg.RollappId)
+	if !found {
+		return nil, types.ErrUnknownRollappID
+	}
+
+	// Get the channel and validate it's connected client chain is the same as the rollapp's
+	_, clientState, err := k.channelKeeper.GetChannelClientState(ctx, "transfer", msg.ChannelId)
+	if err != nil {
+		return nil, err
+	}
+	tmClientState, ok := clientState.(*tenderminttypes.ClientState)
+	if !ok {
+		return nil, sdkerrors.Wrapf(types.ErrInvalidGenesisChannelId, "expected tendermint client state, got %T", clientState)
+	}
+	if tmClientState.GetChainID() != msg.RollappId {
+		return nil, sdkerrors.Wrapf(types.ErrInvalidGenesisChannelId, "channel %s is connected to chain ID %s, expected %s",
+			msg.ChannelId, tmClientState.GetChainID(), msg.RollappId)
+	}
+
+	// Update the rollapp with the channle ID and trigger the genesis event
```

Comment:
```suggestion
	// Update the rollapp with the channel ID and trigger the genesis event
```
Also, does this actually trigger the genesis event, or just processes that part of the event?